src/util/mpir_hwtopo.c

```diff
@@ -462,6 +462,24 @@ MPIR_hwtopo_gid_t MPIR_hwtopo_get_obj_by_name(const char *name)
             hwtopo_class_e class = get_type_class(non_io_ancestor->type);
             gid = HWTOPO_GET_GID(class, non_io_ancestor->depth, non_io_ancestor->logical_index);
         }
+    } else if (!strcmp(name, "bindset")) {
+        char buf[100];          /* covers up to 800 PUs */
```

Comment:
Can you please explain the semantics of `bindset`? We might want to have a detailed description in the commit message. Could you please also update the MPICH wiki for process binding?

Reply:
The "PU" unit info is lost in "bindset" naming. Maybe we want to rename it to something like "puset"?. Actually the "set" format seems suitable for all resource types. E.g., define set of CPUs for each rank.

It looks like the "set" format is not currently supported by hydra `-bind-to` option.  How do you deal with it for the binding operation? I'do see a commit in this PR

Not sure whether it is needed to have "bindset" in the initial impl. I feel that we might need more work to make it complete. Maybe we want to drop it from this PR for now?

Reply:
The commit is dropped, but I'll reply anyway for completeness --

> The "PU" unit info is lost in "bindset" naming. Maybe we want to rename it to something like "puset"?. Actually the "set" format seems suitable for all resource types. E.g., define set of CPUs for each rank.
> 

Probably "cpuset" is a more common name. However, I want to emphasize the "bind" part. Cpuset by itself doesn't mean much. Maybe "cpubindset"?

> It looks like the "set" format is not currently supported by hydra `-bind-to` option. How do you deal with it for the binding operation? I'do see a commit in this PR
> 
It's supported, e.g. `mpiexec -bind-to -user:0,1+2,1+2`.  This PR https://github.com/pmodels/mpich/pull/5095 further extents the syntax so it now supports `mpiexec -bind-to -user:0:1-63x63`